buildSrc/src/main/kotlin/com/datadog/gradle/plugin/instrumentation/DatadogMethodVisitor.kt

```diff
@@ -0,0 +1,23 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.gradle.plugin.instrumentation
+
+import org.objectweb.asm.MethodVisitor
+
+class DatadogMethodVisitor(apiVersion:Int, original:MethodVisitor):
```

Comment:
FYI I don't think you need this another wrapper, you could just directly return `DatadogAttachMethodTransformer` and `DatadogDetachMethodTransformer` from the `DatadogClassVisitor`

Reply:
Hey tks for you input,  I know :). This was just a POC project where I tested several approaches, not meant to be committed in our repo. Thanks for open - sourcing your project, was very helpful for me to get a handle around ASM Plugins.